tests/tests/balances.rs

```diff
@@ -330,3 +330,260 @@ async fn first_5_balances() {
+        base_asset_coin: BaseAssetCoin,
+        chunk_size: i32,
+        base_asset_id_byte: u8,
+    ) {
+        // Given
+
+        // Owner has the following assets:
+        // |   asset    | asset_id  | amount |  type   |         when?          |
+        // | ---------- | --------- | ------ | ------- | ---------------------- |
+        // | asset_1    | 0x2222... | 11     | coin    | always                 |
+        // | asset_2    | 0x7777... | 22     | coin    | always                 |
+        // | base_asset | 0x????... | 33     | coin    | BaseAssetCoin::Present |
+        // | n/a        | 0x????... | 44     | message | MessageCoin::Present   |
+        //
+        // Please note that the lexicographical order of "base asset" is dependent on the test parameter,
+        // so we can check for all three cases, i.e.: base asset is first, last or in the middle
+        // of other assets, like so:
+        // 1) base asset, asset_1, asset_2
+        // 2) asset_1, base_asset, asset_2
+        // 3) asset_1, asset_2, base_asset
+        let base_asset_id = AssetId::from([base_asset_id_byte; 32]);
+
+        let owner = Address::from([0xaa; 32]);
+        let asset_1 = AssetId::new([0x22; 32]);
+        let asset_2 = AssetId::new([0x77; 32]);
+        let mut assets = vec![(asset_1, 11), (asset_2, 22)];
+        if let BaseAssetCoin::Present = base_asset_coin {
+            assets.push((base_asset_id, 33));
+        }
+        let config = setup(
+            &owner,
+            &assets,
+            match message_coin {
+                MessageCoin::Present => Some(MESSAGE_BALANCE),
+                MessageCoin::Missing => None,
+            },
+            base_asset_id,
+        )
+        .await;
+        let srv = FuelService::new_node(config).await.unwrap();
+        let client = FuelClient::from(srv.bound_address);
+
+        // When
+        let mut cursor = None;
+        let mut actual_balances = vec![];
+        loop {
+            let paginated_result = client
+                .balances(
+                    &owner,
+                    PaginationRequest {
+                        cursor,
+                        results: chunk_size,
+                        direction,
+                    },
+                )
+                .await
+                .unwrap();
+            assert!(paginated_result.results.len() <= chunk_size as usize);
+
+            cursor = paginated_result.cursor;
+            actual_balances.extend(
+                paginated_result
+                    .results
+                    .iter()
+                    .map(|r| (r.asset_id, r.amount)),
+            );
+            if !paginated_result.has_next_page {
+                break
+            }
+        }
+
+        // Then
+
+        // Please mind that if present, base asset id is always reported first
+        // (or last in case of backward pagination).
+        let mut expected_balances = match (message_coin, base_asset_coin) {
```

Comment:
[nit] It would be nice to have the `expected_balances` part of the input to the test. But I'm not sure how it will affect readability=)